contracts-periphery/src/UmbraBatchSend.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IUmbra {
+  function sendEth(
+    address payable receiver,
+    uint256 tollCommitment,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+
+  function sendToken(
+    address receiver,
+    address tokenAddr,
+    uint256 amount,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+}
+
+contract UmbraBatchSend {
+  IUmbra internal immutable umbra;
+
+  struct SendEth {
+    address payable receiver;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+
+  struct SendToken {
+    address receiver;
+    address tokenAddr;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+  
+  error ValueMismatch();
+  event BatchSendExecuted(address indexed sender);
+
+  constructor(address umbraAddr) {
+    umbra = IUmbra(umbraAddr);
+  }
+
+  function batchSendEth(uint256 _tollCommitment, SendEth[] calldata _params) external payable {
+    uint256 valueSentAccumulator = _valueSentAccumulator(_tollCommitment, _params);
```

Comment:
The temporary variable here is not adding any value. It doesn't make the code clearer or terse. Let's just use the call to the method directly in the `if` condition below.